kernel/aster-nix/src/process/program_loader/elf/load_elf.rs

```diff
@@ -45,15 +45,7 @@ pub fn load_elf_to_vm(
 ) -> Result<ElfLoadInfo> {
     let parsed_elf = Elf::parse_elf(file_header)?;
 
-    let ldso = if parsed_elf.is_shared_object() {
```

Comment:
Should we only find ldso when the parsed_elf is shared object or executable? And whether the ldso is essential for shared_object?
```rust
let ldso = if parsed_elf.is_executable() {
    lookup_and_parse_ldso(&parsed_elf, file_header, fs_resolver)?;
} else if parsed_elf.is_shared_object() {
    Some(lookup_and_parse_ldso(&parsed_elf, file_header, fs_resolver)?.unwrap())
} else {
    None
};
```

The elf may [six possible types](https://docs.rs/xmas-elf/0.9.1/xmas_elf/header/enum.Type.html#). But I'm not sure whether other 4 types will occur here.

Reply:
It's a little bit noisy and inelegant to check these conditions.
In Linux, only `ET_EXEC` and `ET_DYN` (shared object) ELF can be executed, so it should be guaranteed that `parsed_elf` is shared object or executable when trying to load the interpreter.
And, in Linux, a `.so` without `PT_INTERP` can be executed (although it may fault), so maybe we don't need to ensure its presence in `execve`?

Reply:
Thanks for your explanation! Your statements make sense and the PR looks good to me.

> so it should be guaranteed that parsed_elf is either shared object or executable when trying to load the interpreter.

I have rechecked the code, and this property has been verified. So we don't need to check other elf types here.

https://github.com/asterinas/asterinas/blob/c6aa9f9ee860bbdcb8bb3444b630aff9f0ac14af/kernel/aster-nix/src/process/program_loader/elf/elf_file.rs#L206-L209

> And, in Linux, a .so without PT_INTERP can be executed (although it may fault)

As of now, it's possible that the support for a shared object without a PT_INTERP segment is missing, as we don't have the default interpreter path. However, addressing this issue may be beyond the scope of the current pull request.